code/datums/elements/connect_loc.dm

```diff
@@ -1,63 +1,78 @@
 /// This element hooks a signal onto the loc the current object is on.
 /// When the object moves, it will unhook the signal and rehook it to the new object.
 /datum/element/connect_loc
-	element_flags = ELEMENT_BESPOKE | ELEMENT_DETACH
-	id_arg_index = 2
+	element_flags = ELEMENT_BESPOKE | ELEMENT_DETACH | ELEMENT_COMPLEX_DETACH
+	id_arg_index = 3
 
 	/// An assoc list of signal -> procpath to register to the loc this object is on.
 	var/list/connections
 
 	/// An assoc list of locs that are being occupied and a list of targets that occupy them.
 	var/list/targets = list()
 
-/datum/element/connect_loc/Attach(datum/target, list/connections)
+	/// The callback used when the turf under the tracked object changes
+	var/datum/callback/changeturf_callback
+
+/datum/element/connect_loc/New()
+	. = ..()
+	changeturf_callback = CALLBACK(src, .proc/post_turf_change)
+
+/datum/element/connect_loc/Attach(datum/listener, atom/movable/tracked, list/connections)
```

Comment:
```suggestion
/datum/element/connect_loc/Attach(datum/listener, list/connections, atom/movable/tracked)
```
would it be better to do this and set 
```
tracked = tracked || listener
```
so you wouldnt need the extra argument if listener == tracked

Reply:
Due to how bespoke elements work, no this isn't possible. If you look above the index has changed so that the tracked var is ignored when finding a duplicate element.